zerver/lib/actions.py

```diff
@@ -4104,10 +4104,11 @@ def do_update_user_status(user_profile: UserProfile,
                           away: Optional[bool],
                           status_text: Optional[str],
                           client_id: int) -> None:
-    if away:
-        status = UserStatus.AWAY
+
+    if away is not None:
+        status = away
     else:
-        status = UserStatus.NORMAL
+        status = UserStatus.objects.get(user_profile=user_profile).status
```

Comment:
We don't want to go to the database to get a value here.  The `update_user_status` function should probably just get passed `None` for `status` when `away` is `None`.  It is smart about updating only fields that have changed:

~~~ py
 38 def update_user_status(user_profile_id: int,
 39                        status: Optional[int],
 40                        status_text: Optional[str],
 41                        client_id: int) -> None:
 42
 43     timestamp = timezone_now()
 44
 45     defaults = dict(
 46         client_id=client_id,
 47         timestamp=timestamp,
 48     )
 49
 50     if status is not None:
 51         defaults['status'] = status
 52
 53     if status_text is not None:
 54         defaults['status_text'] = status_text
 55
 56     UserStatus.objects.update_or_create(
 57         user_profile_id=user_profile_id,
 58         defaults=defaults,
 59     )